src/ab-test-preview.js

```diff
@@ -0,0 +1,57 @@
+import { getLetter } from '../src/utils';
+
+document.querySelectorAll( '.ab-test-xb-preview' ).forEach( xb => {
+
+	const xbPostId = xb.getAttribute( 'post-id' );
+	const templates = document.querySelectorAll( 'template[data-parent-id="' + xb.dataset.clientId + '"]' );
+	const tabContainer = xb.querySelector( '.ab-test-xb-preview__tabs' );
+	const tabContent = xb.querySelector( '.ab-test-xb-preview__content' );
+
+	const regex = new RegExp( `(utm_campaign|set_test)=test_xb_${ xbPostId }:(\\d+)`, 'i' );
+	const url_test = unescape( window.location.search ).match( regex );
+
+	for ( let i = 0; i < templates.length; i++ ) {
+		// create a button element
+		const tab = document.createElement( 'button' );
+
+		// set the button inner html to the test title
+		tab.innerHTML = templates[i].dataset.title;
+
+		// if the title is empty the set the title based on the index
+		if ( templates[i].dataset.title === '' ) {
+			tab.innerHTML = 'Variant ' + getLetter( i );
```

Comment:
As far as possible code should be translatable, so you could use the `__()` function to do this. See other JS files for how to include functions from `wp.i18n`.

Reply:
Thanks, this one slipped through. I initially went to do the translate but left it out on as my initial thoughts were that the user will have entered it in the desired language, however the "fallback" title slipped my mind.